### PR TITLE
Code cleanup

### LINKED ISSUE #70: array does not support substitution
Consider this:

```
foo = hello
bar = [${foo}]
```
This results in an exception: 
```
Hocon.HoconParserException : Failed to parse Hocon array. Expected `Comma` or `EndOfLine, found `SubstituteRequired` instead.
```

### LINKED ISSUE #75: Bug when reading array of objects
Could not omit coma when creating an array of objects.

### PR DESCRIPTION
**Pull request by Arkatufus**
- Simplify object parsing.
- Fix end of line parsing bugs that causes object and array parsing.
- Resolves #75 
- Merge and closes #73 pull request
  - Resolves #70 
- Merge and closes #74 pull request
  - Fix problems with CI not running test in their build directory causing it to fail to read included external test files.

### CODE REVIEW

**Comment by Aaronontheweb** [comment]
@Arkatufus unfortunately the file renames I did to make builds run on Linux appear to have created some conflict issues here - best bet is to rebase probably

**Comment by Aaronontheweb** [comment]
CI should play much more nicely now though

**Comment by Arkatufus** [comment]
@Aaronontheweb, Seems like at least one file were not copied from the git repo to the build directory in Linux PR validation build. Build failed on Hocon.Configuration.Test because it could not find the app.config file in the source directory.

File: src/Hocon.Configuration.Test/Hocon.Configuration.Tests.csproj [hunk 1]
```diff
@@ -19,8 +22,12 @@
     <ProjectReference Include="..\Hocon.Configuration\Hocon.Configuration.csproj" />
   </ItemGroup>
 
-  <ItemGroup Condition=" '$(TargetFramework)' == 'net452' ">
+  <ItemGroup Condition=" '$(TargetFramework)' == 'net461' ">
     <Reference Include="System.Configuration" />
   </ItemGroup>
 
+  <Target Name="CopyCustomContent" AfterTargets="AfterBuild">
+    <Copy SourceFiles="app.config" DestinationFiles="$(OutDir)\testhost.dll.config" />
```
Comment (Aaronontheweb) [inline src/Hocon.Configuration.Test/Hocon.Configuration.Tests.csproj]: @Arkatufus change this to `App.config` with a capital A and that should fix the build on Linux

**Pull request merged**